fe/fe-core/src/main/java/org/apache/doris/nereids/util/PlanUtils.java

```diff
@@ -42,12 +43,12 @@ public static Plan projectOrSelf(List<NamedExpression> projectExprs, Plan plan)
         return project(projectExprs, plan).map(Plan.class::cast).orElse(plan);
     }
 
-    public static Optional<LogicalFilter<? extends Plan>> filter(List<Expression> predicates, Plan plan) {
-        return ExpressionUtils.optionalAnd(predicates)
-                .map(predicate -> new LogicalFilter<>(predicate, plan));
+    public static Optional<LogicalFilter<? extends Plan>> filter(Set<Expression> predicates, Plan plan) {
+        return ExpressionUtils.optionalAnd(predicates.toArray(new Expression[0]))
+                .map(opt -> new LogicalFilter<>(predicates, plan));
```

Comment:
why must convert to array? u could add a new function that parameter is a Collection